### ISSUE TITLE
Model input and parallel processing

### ISSUE THREAD

**Issue by Irazall** [mathijsdeen/ClusterBootstrap#12]
Thank you for this awesome package! However, I encountered a minor bug while using `as.formula`. So what happens is, that the method `clusbootglm` does not work properly for models that use the method `as.formula`. If `n.cores` is set to 1, everything works fine, but as soon as you take a higher value, you get the following error: `4 nodes produced errors; first error: object of type 'closure' is not subsettable`

This is an example to reproduce the error:
```
data("medication")
medication$pos_dich <- with(medication, ifelse(pos>quantile(pos,.99),1,0))
formula = "pos_dich ~ treat * time"
model <- glm(formula, medication, family="binomial")
clusbootglm(model = model, data = medication, clusterid = id, family = binomial, n.cores=2, B=100)
model <- glm(as.formula(formula), medication, family="binomial")
clusbootglm(model = model, data = medication, clusterid = id, family = binomial, n.cores=1, B=100)
clusbootglm(model = model, data = medication, clusterid = id, family = binomial, n.cores=2, B=100)
```

All function calls except the last one work fine. My suggestion is either to fix this somehow or at least mention this in documentation on CRAN

**Comment by mathijsdeen** [comment]
Hi Irazall,

First of all, thank you for your kind words, and for using the ClusterBootstrap package! And, of course, for the effort of bringing this to my attention. 

It seems that you are using a glm object as input for the `model` argument. When I first looked at your code, I was surprised to see that this approach works at all for the single CPU core case. 

I think this is a case of "dubious documentation", for which I am to blame. As you can see in the example in the help file for the `clusbootglm` function and from the examples given in our paper (https://link.springer.com/article/10.3758/s13428-019-01252-y), the `model` argument is a formula on every occasion. This was also what it's intended to be. I realize that it would be more logical if the `model` argument was named `formula` instead. Sadly, for the sake of backward compatibility, the argument name cannot be changed. Furthermore, the description of the `model` argument in the help file should be clearer about the fact that this should be a formula - at the moment, it merely states "generalized linear model to be fitted with the cluster bootstrap" which is not clear at all and could easily be interpreted as "this needs to be a glm class object". 

So, the fix on the user end is to use a formula instead. Intriguingly, on the last line, "model = as.formula(model)" seems to work fine - although I do not yet know why that does work. It seems something wonky is going on when the `clusbootglm` function passes some local variables to the cluster array in `parallel` and that needs some straightening out.

What I will do for the next update is the following:

- I will make the use of a glm class object as input for the `model` argument work for the multiple core case. The `model` argument then can handle a formula class object as well as a glm class object. Documentation will be updated accordingly.
- If the attempt for the point above fails, the need for a formula will be stated more explicitly in the documentation, at least for the multiple core case.

Once again, thanks for bringing this up! Please let me know if you run into any other issues.

Best,
Mathijs

**Comment by mathijsdeen** [comment]
This issue is solved in version 1.1.1, `glm` objects can be used with parallel computing as well. 

Version 1.1.1 is not yet available on CRAN, use devtools::install_github() to install this version.
